### DISCUSSION TITLE
[Config Support]: Motion detection for light changes and IR

### DISCUSSION THREAD

**Discussion by lordraiden** [blakeblackshear/frigate#12267] [Config Support]
### Describe the problem you are having

As seen in the video the process is the light is on, goes off, IR camera gets activated, the process will continue with light on, camera IR disable. etc

I have tried everything but still frigate detects this as motion, Is there a way to fix it? any advice?

I'm using this parameters:

### Version

System 0.13.2-6476f8a

### Frigate config file

```yaml
motion:
      mask:
        - 903,65,1664,65,1664,0,903,0 # timestamp reolink
        - 681,1234,681,1276,629,1276,629,1234 # elevator numbers
      threshold: 200
      contour_area: 30
      lightning_threshold: 0.3
```

### Relevant log output

```shell
N/A
```

### Operating system

UNRAID

### Install method

Docker Compose

### Object Detector

CPU (no coral)

**Comment by NickM-27** [comment]
Motion is pixels changing, in this case almost every single pixel is changing. If that is not detected as motion then nothing will be. Is there a reason why you need this to not be detected as motion?

**Reply by lordraiden** [reply]
> Motion is pixels changing, in this case almost every single pixel is changing. If that is not detected as motion then nothing will be. Is there a reason why you need this to not be detected as motion?

I was trying to avoid the use of detect to lower the CPU consumption.
I was hopping that motion "lightning_threshold" setting could be used to filter out those full changes (light on/off, IR on/off).

https://docs.frigate.video/configuration/motion_detection/#tuning-for-large-changes-in-motion

> **Large changes in motion like PTZ moves and camera switches between Color and IR mode should result in no motion detection.** This is done via the lightning_threshold

I guess is not working as intended even in the lowest level available 0.3

Basically I need object motion but I don't need the object to be identify and do this with the lower CPU consumption possible.

But I guess If there is no alternative I will use detect as well

**Reply by NickM-27** [reply]
lightning_threshold will cause the camera to recalibrate. It does not eliminate all possible motion when these changes happen. 

I don't think the goal of 0 motion detection really matters, a couple of seconds of motion detection is not going to cause crazy CPU usage.

**Reply by lordraiden** [reply]
> lightning_threshold will cause the camera to recalibrate. It does not eliminate all possible motion when these changes happen.
> 
> I don't think the goal of 0 motion detection really matters, a couple of seconds of motion detection is not going to cause crazy CPU usage.

the thing is I use it as an "alarm" system so I don't want to record a video and get an alert every time the light changes.
On the other hand I'm not worried about the CPU consumption while using motion, I am worried that the only solution to this is to enable detect, If I use detect it will consume CPU even if there is no motion at all, at least in my case.

**Reply by NickM-27** [reply]
Right, frigate is built to use object detection really.

> If I use detect it will consume CPU even if there is no motion at all, at least in my case.

Why would it use CPU even when no motion is detected?

**Reply by lordraiden** [reply]
> Right, frigate is built to use object detection really.
> 
> > If I use detect it will consume CPU even if there is no motion at all, at least in my case.
> 
> Why would it use CPU even when no motion is detected?

Well... I think that motion was poorly calibrated so detect was being triggered constantly, anyway is my fault, thanks for helping me out here. I will calibrate first motion properly.

**Reply by lordraiden** [reply]
It looks like that was the problem.

Now the thing is, is there a way to detect generic objects? like a full set of pixels that are together moving, but not necessary any object in particular.

**Reply by NickM-27** [reply]
No, that is basically just describing more advanced motion detection. Object detection detects specific objects.In this case would you have a reason to expect anything other than a person or dog to be detected though?

**Reply by lordraiden** [reply]
> No, that is basically just describing more advanced motion detection. Object detection detects specific objects.In this case would you have a reason to expect anything other than a person or dog to be detected though?

Not necessarily but these are zones that there shouldn't be any object moving, I don't care what they are if they are  detected as people or not, imagine that there is something moving but is not detected as a person for whatever reason but is actually a person moving with a customer or hiding behind in a plank...
Any way I have lower the threshold and min score for person detection we will see.

**Reply by NickM-27** [reply]
the new ui in 0.14 has object review as well as a tab for reviewing motion that was detected without an associated object

**Reply by Kedryn** [reply]
> Motion is pixels changing, in this case almost every single pixel is changing. If that is not detected as motion then nothing will be. Is there a reason why you need this to not be detected as motion?

Blue Iris has a settings that is specifically made for this problem. If more than x% of the scene changes, than there is no motion detected.
It is made for people that has, for example, the light source inside the camera view.

I think it's a good thing to add to Frigate.
I have 2 cams (garage and outside the main door) that receives light when motion is detected in the night, and yet i didn't manage to NOT have frigate detect motion when the light goes off and IR kicks in.

Without that setting, i think i'll have to DISABLE motion for like 2-3 secs before switching off the light source.

**Reply by NickM-27** [reply]
Or you could just have a cool down before the light will turn back on again. Frigates motion detection is not as sophisticated as other system which lack intelligent object tracking. Triggering off an object is what I do and it works perfectly

You can also use lightning threshold as that is what the option is specifically for

**Reply by Kedryn** [reply]
> Or you could just have a cool down before the light will turn back on again. Frigates motion detection is not as sophisticated as other system which lack intelligent object tracking. Triggering off an object is what I do and it works perfectly
> 
> You can also use lightning threshold as that is what the option is specifically for

What a "cool down" will solve? When the light turns off, a movement is detected.

I already tested lightning threshold to 0.3 (minium value accepted), but it does not solve the problem.

**Reply by NickM-27** [reply]
basically, once the lights turn off you require a few seconds before the lights can turn on again. You are using motion detection in a way it is not designed to be used for in frigate

**Reply by Kedryn** [reply]
> basically, once the lights turn off you require a few seconds before the lights can turn on again. You are using motion detection in a way it is not designed to be used for in frigate

Uhm, no, that's not what i'm trying to explain.
Let me try again.

I have my main entrance like in this photo:
![image](https://github.com/user-attachments/assets/6ed60fb7-f7c0-4100-a168-810ad2d08f3a)

on the top left you can see a light bulb.
When movemente is detected, after sunset, the light turns on.
After 3 minutes, the lights turns off.
When the lights turn off, there is a massive motion detected, like 100%, that turns it on again.
The "lightining_threshold" can't go lower than 0.3, so it's not enough.
Before switching to Frigate, i was using (and still i'm, but only for scheduled recordings), Blue Iris. 
In BI there is an option that skips motion detection when it is greater of a chosen percentage (i use 80%), and it works flawlessy in my situation.

The same is for the garage,
With Frigate, right now, my lights turns on at first movement detected, and turns off at sunrise, with a glitch every 3 minutes.

**Reply by NickM-27** [reply]
Right, I'm telling you that you should change the logic so that it won't turn on the light immediately after it turns off. 

For frigate the detection runs at 5 fps, it's entirely possible that that first frame after the light turns on doesn't have > 30% of the pixels changing yet so motion is detected then a frame or two later the lightning detection kicks in. That would explain why you still get motion detection triggering. You can fix it the way I described.

**Reply by Kedryn** [reply]
being simpler for my setup with home assistant, i tried to disable the motion and detection for like 8 seconds BEFORE turning off the light.

Turn motion/detection off->turn light off->wait 8 seconds->Turno motion/detection on

When i turn motion and detection on again, frigate signals motion and the light turns on again.

I'll try the "cool down" route for the light on

**Reply by amd2800barton** [reply]
I stumbled on this thread while I was looking for ways to transfer my Reolink cameras from Unifi Protect to Frigate. The way I solved my "lights off triggers motion triggers lights on" issue is that I have my cameras connected to home assistant (no video stream, detection events and settings control only), and have HA do the automation. My HA logic is:

No motion for 30 seconds -> switch camera to night vision mode and disable the motion lights on trigger -> wait 5 seconds for night vision to adjust from color vision -> trigger sconce light off -> wait 5 seconds ->re-enable motion lights trigger

On sunrise I have an automation that disables the motion light, and sets the camera to color/auto. On sunset the camera gets set to night vision and the motion trigger gets turned on 5 seconds later.

Haven't figured out how to do that in Frigate yet, so I might just be leaving the logic in HA but switching from the Reolink motion/person detection to Frigate.
